client/src/main/java/org/apache/celeborn/client/write/SlowStartPushSpeedStrategy.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.client.write;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.celeborn.common.CelebornConf;
+
+public class SlowStartPushSpeedStrategy extends PushSpeedStrategy {
+
+  private static final Logger logger = LoggerFactory.getLogger(SlowStartPushSpeedStrategy.class);
+  private int maxInFlight;
+  private final AtomicInteger currentMaxReqsInFlight;
+
+  // Indicate the number of congested times even after the in flight requests reduced to 1
+  private final AtomicInteger continueCongestedNumber;
+  private int congestionAvoidanceFlag = 0;
+
+  private final long maxSleepMills;
+
+  public SlowStartPushSpeedStrategy(CelebornConf conf) {
+    super(conf);
+    this.currentMaxReqsInFlight = new AtomicInteger(0);
+    this.continueCongestedNumber = new AtomicInteger(0);
+    this.maxInFlight = conf.pushMaxReqsInFlight();
+    this.maxSleepMills = conf.pushSlowStartMaxSleepMills();
+  }
+
+  /**
+   * If `pushDataSlowStart` is enabled, will increase `currentMaxReqsInFlight` gradually to meet the
+   * max push speed.
+   *
+   * <p>1. slow start period: every RTT period, `currentMaxReqsInFlight` is multiplied.
```

Comment:
multiplied by some value?

Reply:
will change, here means `currentMaxReqsInFlight` is doubled